$NetBSD: patch-ag,v 1.7 2022/07/27 18:38:48 wiz Exp $

Add DragonFly support.
Fix png library name.
Fix linking against tiff.

--- configure.orig	2004-09-23 01:15:44.000000000 +0000
+++ configure
@@ -3434,13 +3434,13 @@ darwin* | rhapsody*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
-freebsd* | kfreebsd*-gnu)
+freebsd* | kfreebsd*-gnu | dragonfly*)
   if echo __ELF__ | $CC -E - | grep __ELF__ > /dev/null; then
     case $host_cpu in
     i*86 )
       # Not sure whether the presence of OpenBSD here was a mistake.
       # Let's accept both of them until this is cleared up.
-      lt_cv_deplibs_check_method='file_magic (FreeBSD|OpenBSD)/i[3-9]86 (compact )?demand paged shared library'
+      lt_cv_deplibs_check_method='file_magic (DragonFly|FreeBSD|OpenBSD)/i[3-9]86 (compact )?demand paged shared library'
       lt_cv_file_magic_cmd=/usr/bin/file
       lt_cv_file_magic_test_file=`echo /usr/lib/libc.so.*`
       ;;
@@ -7094,7 +7094,7 @@ if test -z "$aix_libpath"; then aix_libp
       ;;
 
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-    freebsd* | kfreebsd*-gnu)
+    freebsd* | kfreebsd*-gnu | dragonfly*)
       archive_cmds='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
       hardcode_libdir_flag_spec='-R$libdir'
       hardcode_direct=yes
@@ -7672,7 +7672,7 @@ kfreebsd*-gnu)
   dynamic_linker='GNU ld.so'
   ;;
 
-freebsd*)
+freebsd* | dragonfly*)
   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
   version_type=freebsd-$objformat
   case $version_type in
@@ -10055,7 +10055,7 @@ if test -z "$aix_libpath"; then aix_libp
   freebsd-elf*)
     archive_cmds_need_lc_CXX=no
     ;;
-  freebsd* | kfreebsd*-gnu)
+  freebsd* | kfreebsd*-gnu | dragonfly*)
     # FreeBSD 3 and later use GNU C++ and GNU ld with standard ELF
     # conventions
     ld_shlibs_CXX=yes
@@ -10762,7 +10762,7 @@ echo $ECHO_N "checking for $compiler opt
 	    ;;
 	esac
 	;;
-      freebsd* | kfreebsd*-gnu)
+      freebsd* | kfreebsd*-gnu | dragonfly*)
 	# FreeBSD uses GNU C++
 	;;
       hpux9* | hpux10* | hpux11*)
@@ -11376,7 +11376,7 @@ kfreebsd*-gnu)
   dynamic_linker='GNU ld.so'
   ;;
 
-freebsd*)
+freebsd* | dragonfly*)
   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
   version_type=freebsd-$objformat
   case $version_type in
@@ -13951,7 +13951,7 @@ if test -z "$aix_libpath"; then aix_libp
       ;;
 
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-    freebsd* | kfreebsd*-gnu)
+    freebsd* | kfreebsd*-gnu | dragonfly*)
       archive_cmds_F77='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
       hardcode_libdir_flag_spec_F77='-R$libdir'
       hardcode_direct_F77=yes
@@ -14529,7 +14529,7 @@ kfreebsd*-gnu)
   dynamic_linker='GNU ld.so'
   ;;
 
-freebsd*)
+freebsd* | dragonfly*)
   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
   version_type=freebsd-$objformat
   case $version_type in
@@ -16309,7 +16309,7 @@ if test -z "$aix_libpath"; then aix_libp
       ;;
 
     # FreeBSD 3 and greater uses gcc -shared to do shared libraries.
-    freebsd* | kfreebsd*-gnu)
+    freebsd* | kfreebsd*-gnu  | dragonfly*)
       archive_cmds_GCJ='$CC -shared -o $lib $libobjs $deplibs $compiler_flags'
       hardcode_libdir_flag_spec_GCJ='-R$libdir'
       hardcode_direct_GCJ=yes
@@ -16887,7 +16887,7 @@ kfreebsd*-gnu)
   dynamic_linker='GNU ld.so'
   ;;
 
-freebsd*)
+freebsd* | dragonfly*)
   objformat=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
   version_type=freebsd-$objformat
   case $version_type in
@@ -23645,8 +23645,8 @@ echo "${ECHO_T}$ac_cv_header_tiffio_h" >
 
 fi
 if test $ac_cv_header_tiffio_h = yes; then
-  TIFFLIBS="-ltiff"
-    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff";
+  TIFFLIBS="-ltiff -ljpeg -lz"
+    SUPPORT_LIBS="$SUPPORT_LIBS -ltiff -ljpeg -lz";
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIBTIFF 1
 _ACEOF
@@ -24395,13 +24395,13 @@ fi
 
 
 
-echo "$as_me:$LINENO: checking for png_read_info in -lpng" >&5
-echo $ECHO_N "checking for png_read_info in -lpng... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for png_read_info in -lpng16" >&5
+echo $ECHO_N "checking for png_read_info in -lpng16... $ECHO_C" >&6
 if test "${ac_cv_lib_png_png_read_info+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lpng -lz -lm $GX_LIBS $LIBS"
+LIBS="-lpng16 -lz -lm $GX_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -24670,9 +24670,9 @@ echo "$as_me: error: *** PNG header file
 fi
 
 if test "$png_ok" = yes; then
-	PNGLIBS="-lpng -lz"
+	PNGLIBS="-lpng16 -lz"
 
-	SUPPORT_LIBS="$SUPPORT_LIBS -lpng -lz";
+	SUPPORT_LIBS="$SUPPORT_LIBS -lpng16 -lz";
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIBPNG 1
 _ACEOF
